examples/jsm/renderers/nodes/accessors/CameraNode.js

```diff
@@ -0,0 +1,83 @@
+import Node from '../core/Node.js';
+import Vector3Node from '../inputs/Vector3Node.js';
+import Matrix4Node from '../inputs/Matrix4Node.js';
+import { NodeUpdateType } from '../core/constants.js';
+
+class CameraNode extends Node {
+
+	static POSITION = 'position';
+	static PROJECTION = 'projection';
+	static VIEW = 'view';
+
+	constructor( scope = CameraNode.POSITION ) {
+
+		super();
+
+		this.updateType = NodeUpdateType.Frame;
+
+		this.scope = scope;
+		
+		this.inputNode = null;
+
+	}
+
+	getType() {
+
+		if ( this.scope === CameraNode.PROJECTION || this.scope === CameraNode.VIEW ) {
+			
+			return 'mat4';
+			
+		}
+
+		return 'vec3';
+
+	}
+	
+	update( frame ) {
+
+		const camera = frame.camera;
+		const inputNode = this.inputNode;
+
+		if ( this.scope === CameraNode.PROJECTION ) {
+			
+			inputNode.value = camera.projectionMatrix;
+			
+		} else if ( this.scope === CameraNode.VIEW ) {
+			
+			inputNode.value = camera.matrixWorldInverse;
+			
+		} else if ( this.scope === CameraNode.POSITION ) {
+			
+			camera.getWorldPosition( inputNode.value );
+			
+		}
+
+	}
+
+	generate( builder, output ) {
+
+		const nodeData = builder.getDataFromNode( this );
+
+		if ( this.initialized !== true ) {
```

Comment:
I see the usage of the `initialized` property for the first time in a node class.

The code section implies that once a scope (`POSITION`, `PROJECTION` or `VIEW`) is defined, it can't be change afterwards, right?

Do you mind defining `initialized` in the constructor? If you plan it to use it elsewhere, it might make sense to define it in `Node`.

Reply:
> The code section implies that once a scope (POSITION, PROJECTION or VIEW) is defined, it can't be change afterwards, right?

`initialized` is changed to `undefined` every time that the shader is compiled... 
this.inputNode is a private node I think that I can rename to `this,_inputNode`?

Reply:
Sounds good!